.github/workflows/build-publish-documentation.yaml

```diff
@@ -18,40 +25,30 @@ jobs:
       #       check-out repo and set-up python
       #----------------------------------------------
       - name: Check out repository
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.2.2
         with:
           fetch-depth: 0 # fetch all commits/branches so that mike works
 
-      - name: Setup Python 3.9
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.9
-              
       #----------------------------------------------
       #          install & configure poetry
       #----------------------------------------------
       - name: Install Poetry
-        uses: snok/install-poetry@v1.3
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
+        run: |
+          pipx install poetry
+          pipx inject poetry poetry-dynamic-versioning
```

Comment:
The repo is not yet setup for dynamic-versioning so you don't need to install it.